modules/querier/http.go

```diff
@@ -100,6 +100,71 @@ func (q *Querier) TraceByIDHandler(w http.ResponseWriter, r *http.Request) {
 	w.Header().Set(api.HeaderContentType, api.HeaderAcceptJSON)
 }
 
+func (q *Querier) TraceByIDHandlerV2(w http.ResponseWriter, r *http.Request) {
+	// Enforce the query timeout while querying backends
+	ctx, cancel := context.WithDeadline(r.Context(), time.Now().Add(q.cfg.TraceByID.QueryTimeout))
+	defer cancel()
+
+	span, ctx := opentracing.StartSpanFromContext(ctx, "Querier.TraceByIDHandler")
+	defer span.Finish()
+
+	byteID, err := api.ParseTraceID(r)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	// validate request
+	blockStart, blockEnd, queryMode, timeStart, timeEnd, err := api.ValidateAndSanitizeRequest(r)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+	span.LogFields(
+		ot_log.String("msg", "validated request"),
+		ot_log.String("blockStart", blockStart),
+		ot_log.String("blockEnd", blockEnd),
+		ot_log.String("queryMode", queryMode),
+		ot_log.String("timeStart", fmt.Sprint(timeStart)),
+		ot_log.String("timeEnd", fmt.Sprint(timeEnd)))
+
+	resp, err := q.FindTraceByID(ctx, &tempopb.TraceByIDRequest{
+		TraceID:    byteID,
+		BlockStart: blockStart,
+		BlockEnd:   blockEnd,
+		QueryMode:  queryMode,
+	}, timeStart, timeEnd)
+	if err != nil {
+		handleError(w, err)
+		return
+	}
+
+	if r.Header.Get(api.HeaderAccept) == api.HeaderAcceptProtobuf {
```

Comment:
i think we have a method that handles some of this:

`writeFormattedContentForRequest`

Reply:
Oh, how I didn't notice? 🙀